src/BenchmarksApps/TLS/HttpSys/NetShWrapper.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Diagnostics;
+using System.Security.Cryptography.X509Certificates;
+
+namespace HttpSys
+{
+    public static class NetShWrapper
+    {
+        public static void DisableHttpSysMutualTls(string ipPort)
+        {
+            Console.WriteLine("Disabling mTLS for http.sys");
+
+            string command = $"http delete sslcert ipport={ipPort}";
+            ExecuteNetShCommand(command);
+
+            Console.WriteLine("Disabled http.sys settings for mTLS");
+        }
+
+        public static void Show()
+        {
+            ExecuteNetShCommand("http show sslcert", alwaysLogOutput: true);
+        }
+
+        public static void EnableHttpSysMutualTls(string ipPort)
+        {
+            Console.WriteLine("Setting up mTLS for http.sys");
+
+            var certificate = LoadCertificate();
+            Console.WriteLine("Loaded `testCert.pfx` from local file system");
+            using (var store = new X509Store(StoreName.My, StoreLocation.LocalMachine))
+            {
+                store.Open(OpenFlags.ReadWrite);
+                store.Add(certificate);
+                Console.WriteLine("Added `testCert.pfx` to localMachine cert store");
+                store.Close();
+            }
+
+            string certThumbprint = certificate.Thumbprint;
+            string appId = Guid.NewGuid().ToString();
+
+            string command = $"http add sslcert ipport={ipPort} certstorename=MY certhash={certThumbprint} appid={{{appId}}} clientcertnegotiation=enable";
+            ExecuteNetShCommand(command);
+
+            Console.WriteLine("Configured http.sys settings for mTLS");
+        }
+
+        private static void ExecuteNetShCommand(string command, bool alwaysLogOutput = false)
+        {
+            ProcessStartInfo processInfo = new ProcessStartInfo("netsh", command)
+            {
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                UseShellExecute = false,
+                CreateNoWindow = true
+            };
+
+            Console.WriteLine($"Executing command: `netsh {command}`");
+            using Process process = Process.Start(processInfo)!;
+            string output = process.StandardOutput.ReadToEnd();
+            process.WaitForExit();
+
+            if (alwaysLogOutput)
+            {
+                Console.WriteLine(output);
+            }
+
+            if (process.ExitCode != 0)
+            {
+                throw new InvalidOperationException($"netsh command execution failure: {output}");
+            }
+        }
+
+#pragma warning disable SYSLIB0057 // Type or member is obsolete
```

Comment:
Can we use the non-obsolete APIs instead?
https://learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0057#workaround

Reply:
I tried those APIs, but they are not allowing loading `pfx` file. I.e. [X509CertificateLoader.LoadCertificateFromFile](https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509certificateloader.loadcertificatefromfile?view=net-9.0) loads a single X.509 certificate (in either the **PEM** or **DER** encoding) from the specified file (**not PFX**).

do u know if there is a non-obsolete API which allows it?

Reply:
https://learn.microsoft.com/dotnet/api/system.security.cryptography.x509certificates.x509certificateloader.loadpkcs12?view=net-9.0 says it loads a pfx. If that doesn't work we can ask someone from runtime.

Reply:
Should be able to remove pragmas now?

Reply:
changed to new API, removed pragmas, thanks for suggestion!